keras/applications/imagenet_utils.py

```diff
@@ -1,33 +1,22 @@
 import json
 import warnings
+import numpy as np
 
 from ..utils.data_utils import get_file
 from .. import backend as K
 
 CLASS_INDEX = None
 CLASS_INDEX_PATH = 'https://s3.amazonaws.com/deep-learning-models/image-models/imagenet_class_index.json'
 
+# Global Numpy arrays of imagenet mean for preprocessing symbolic inputs
+_IMAGENET_MEAN = None
 
-def preprocess_input(x, data_format=None, mode='caffe'):
-    """Preprocesses a tensor encoding a batch of images.
 
-    # Arguments
-        x: input Numpy tensor, 4D.
-        data_format: data format of the image tensor.
-        mode: One of "caffe", "tf".
-            - caffe: will convert the images from RGB to BGR,
-                then will zero-center each color channel with
-                respect to the ImageNet dataset,
-                without scaling.
-            - tf: will scale pixels between -1 and 1,
-                sample-wise.
-
-    # Returns
-        Preprocessed tensor.
-    """
+def _preprocess_numpy_input(x, data_format, mode):
     if data_format is None:
         data_format = K.image_data_format()
-    assert data_format in {'channels_last', 'channels_first'}
+    if data_format not in {'channels_first', 'channels_last'}:
+        raise ValueError('Unknown data_format ' + str(data_format))
```

Comment:
Move this check (as well as setting of default value) to the parent `preprocess_input`